src/Core/Stock/StockManager.php

```diff
@@ -42,12 +49,16 @@ class StockManager
      */
     public function updatePackQuantity($product, $stock_available, $delta_quantity, $id_shop = null)
     {
-        $configuration = \PrestaShop\PrestaShop\Adapter\ServiceLocator::get('\\PrestaShop\\PrestaShop\\Core\\ConfigurationInterface');
+        $serviceLocator = new ServiceLocator();
+
+        $configuration = $serviceLocator::get('\\PrestaShop\\PrestaShop\\Core\\ConfigurationInterface');
```

Comment:
Can you please add a ToDo here, saying we should check where these functions are called?
We should call the needed classes with the Symfony dependency injection instead of the Homemade Service Locator